### DISCUSSION TITLE
How should I customize the rendering of Markdown

### DISCUSSION THREAD

**Discussion by SharpIceX** [mkdocs/mkdocs#3360] [Q&A] [answered]
I hope to
```
![Text](URL)
```
renders as
```html
<a href="URL" data-fancybox="gallery" style="Images">
     <img class="lazy" data-src="URL" decoding="async">
     <p>Text</p>
</a>
```
But I still haven't found a solution after consulting Mkdocs documentation

**My English is not very good, I use Google Translate**

**Comment by pawamoy** [comment, marked as answer]
Sorry you didn't get help earlier. I won't be super helpful either, but you might want to check out how to write extensions for Python-Markdown here: https://python-markdown.github.io/extensions/api/. You're probably looking at writing an inline processor: https://python-markdown.github.io/extensions/api/#inlineprocessors.

**Reply by SharpIceX** [reply]
Ah, thank you for your help. I have actually completed my needs through the ```on_page_markdown``` hook of the mkdocs plugin and ```re.sub```

**Reply by pawamoy** [reply]
Great, that works too!
